amnahhashmi/shield
# Incorporate Celery for task queues

To Do:
- [x] Update docs to get started (Installing Redis, etc.)
- [x] Add to Procfile
- [x] :bug: Fix hang on async task scheduler
- [x] Move URLs for Celery into env vars

Test Plan:
1. Install new packages: `pip install -r requirements.txt`
2. Install [Redis](https://redis.io/topics/quickstart) on your machine
3. Start Redis server with default settings
4. Set REDIS_URL environment variable to point to your local Redis server
5. Start Celery worker: `celery -A shield worker --pool=solo -l info`
6. In another terminal: `python manage.py shell`
7. `from errand_matcher.tasks import add`
8. `res = add.delay(4, 4)`
9. `res.ready()` should return True. If False, repeat until value reads True
10. `res.get()` should spit out 8 to the terminal

Questions:
- How are we going to deploy this? Should we fire up another dyno to process the task queue?
- How do we want to deploy Redis? Assuming a small load, existing dynos would be fine, but worth thinking about if we expect load to increase significantly.
